Zend/zend_language_parser.y

```diff
@@ -86,145 +86,145 @@ static YYSIZE_T zend_yytnamerr(char*, const char*);
-%token <ident> T_FOREACH       "foreach (T_FOREACH)"
-%token <ident> T_ENDFOREACH    "endforeach (T_ENDFOREACH)"
-%token <ident> T_DECLARE       "declare (T_DECLARE)"
-%token <ident> T_ENDDECLARE    "enddeclare (T_ENDDECLARE)"
-%token <ident> T_AS            "as (T_AS)"
-%token <ident> T_SWITCH        "switch (T_SWITCH)"
-%token <ident> T_ENDSWITCH     "endswitch (T_ENDSWITCH)"
-%token <ident> T_CASE          "case (T_CASE)"
-%token <ident> T_DEFAULT       "default (T_DEFAULT)"
-%token <ident> T_BREAK         "break (T_BREAK)"
-%token <ident> T_CONTINUE      "continue (T_CONTINUE)"
-%token <ident> T_GOTO          "goto (T_GOTO)"
-%token <ident> T_FUNCTION      "function (T_FUNCTION)"
-%token <ident> T_FN            "fn (T_FN)"
-%token <ident> T_CONST         "const (T_CONST)"
-%token <ident> T_RETURN        "return (T_RETURN)"
-%token <ident> T_TRY           "try (T_TRY)"
-%token <ident> T_CATCH         "catch (T_CATCH)"
-%token <ident> T_FINALLY       "finally (T_FINALLY)"
-%token <ident> T_THROW         "throw (T_THROW)"
-%token <ident> T_USE           "use (T_USE)"
-%token <ident> T_INSTEADOF     "insteadof (T_INSTEADOF)"
-%token <ident> T_GLOBAL        "global (T_GLOBAL)"
-%token <ident> T_STATIC        "static (T_STATIC)"
-%token <ident> T_ABSTRACT      "abstract (T_ABSTRACT)"
-%token <ident> T_FINAL         "final (T_FINAL)"
-%token <ident> T_PRIVATE       "private (T_PRIVATE)"
-%token <ident> T_PROTECTED     "protected (T_PROTECTED)"
-%token <ident> T_PUBLIC        "public (T_PUBLIC)"
-%token <ident> T_VAR           "var (T_VAR)"
-%token <ident> T_UNSET         "unset (T_UNSET)"
-%token <ident> T_ISSET         "isset (T_ISSET)"
-%token <ident> T_EMPTY         "empty (T_EMPTY)"
-%token <ident> T_HALT_COMPILER "__halt_compiler (T_HALT_COMPILER)"
-%token <ident> T_CLASS         "class (T_CLASS)"
-%token <ident> T_TRAIT         "trait (T_TRAIT)"
-%token <ident> T_INTERFACE     "interface (T_INTERFACE)"
-%token <ident> T_EXTENDS       "extends (T_EXTENDS)"
-%token <ident> T_IMPLEMENTS    "implements (T_IMPLEMENTS)"
-%token <ident> T_NAMESPACE     "namespace (T_NAMESPACE)"
-%token <ident> T_LIST            "list (T_LIST)"
-%token <ident> T_ARRAY           "array (T_ARRAY)"
-%token <ident> T_CALLABLE        "callable (T_CALLABLE)"
-%token <ident> T_LINE            "__LINE__ (T_LINE)"
-%token <ident> T_FILE            "__FILE__ (T_FILE)"
-%token <ident> T_DIR             "__DIR__ (T_DIR)"
-%token <ident> T_CLASS_C         "__CLASS__ (T_CLASS_C)"
-%token <ident> T_TRAIT_C         "__TRAIT__ (T_TRAIT_C)"
-%token <ident> T_METHOD_C        "__METHOD__ (T_METHOD_C)"
-%token <ident> T_FUNC_C          "__FUNCTION__ (T_FUNC_C)"
-%token <ident> T_NS_C            "__NAMESPACE__ (T_NS_C)"
+%token <ast> T_ENCAPSED_AND_WHITESPACE  "quoted string and whitespace"
+%token <ast> T_CONSTANT_ENCAPSED_STRING "quoted string"
+%token <ast> T_STRING_VARNAME "variable name"
+%token <ast> T_NUM_STRING "number"
+
+%token <ident> T_INCLUDE       "'include'"
```

Comment:
Our preference is to use double quotes in error messages.

Reply:
The current output uses single quotes in this context, e.g. https://3v4l.org/rnKJ3 (`unexpected 'echo' (T_ECHO)`)

Should be easy enough to change if it makes it more consistent with other types of message though.

Reply:
I have a PR where we're trying to get rid of `'` in favour of `"`, so once it's merged, `"` will be more the consistent choice. :)

Reply:
OK, I'll see if I can get it to show `"` without descending too far into backslash-escape hell :)

Reply:
It turned out to be easiest to leave the strings here with single quotes to mark them as literal tokens, then detect and substitute in the formatting routine (which is only run after a parse error, so doesn't need to be highly performant), both because of the odd way the generated code handles backslash-escapes, and the fact that single-character tokens come through in single quotes anyway.